app/controllers/discussions_messages_controller.rb

```diff
@@ -0,0 +1,24 @@
+class DiscussionsMessagesController < AjaxController
+  before_action :set_discussion!, only: [:create, :destroy]
+
+  def create
+    @discussion.submit_message! message_params, current_user
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy
+    message = Message.find(params[:id])
+    message.destroy! if message.authorized?(current_user)
+    redirect_back(fallback_location: root_path)
```

Comment:
What happens if destroy action is not authorized? Is there any feedback given to the user? 

If this should never happen, I prefer to simply raise an Unauthorized exception